File: infrastructure/adapters/in/PatientController.js
// Controlador REST para pacientes
const express = require('express');

function createPatientRouter(patientService) {
    const router = express.Router();

    // POST - para registrar un paciente
    router.post('/', (req, res) => {
        const { nombre, email } = req.body;
        if (!nombre || !email) {
            return res.status(400).json({ error: 'nombre y email son obligatorios' });
        }
        const patient = patientService.registerPatient(nombre, email);
        res.status(201).json(patient);
    });

    // GET - para consultar todos los pacientes
    router.get('/', (req, res) => {
        const patients = patientService.getAllPatients();
        res.json(patients);
    });

    // GET - consultar paciente por id
    router.get('/:id', (req, res) => {
        const patient = patientService.getPatientById(Number(req.params.id));
        if (!patient) {
            return res.status(404).json({ error: 'Paciente no encontrado' });
        }
        res.json(patient);
    });

    // DELETE - para eliminar un paciente
    router.delete('/:id', (req, res) => {
        const deleted = patientService.deletePatient(Number(req.params.id));
        if (!deleted) {
            return res.status(404).json({ error: 'Paciente no encontrado' });
        }
        res.json({ message: 'Paciente eliminado correctamente' });
    });

    return router;
}

module.exports = createPatientRouter;

File: infrastructure/adapters/in/patient_controller.py
# Controlador REST para pacientes
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel

class PatientRequest(BaseModel):
    nombre: str
    email: str

def create_patient_router(patient_service):
    router = APIRouter()

    # POST - para registrar un paciente
    @router.post("/patients", status_code=201)
    def register_patient(data: PatientRequest):
        patient = patient_service.register_patient(data.nombre, data.email)
        return patient.to_dict()

    # GET - para consultar todos los pacientes
    @router.get("/patients")
    def get_all_patients():
        patients = patient_service.get_all_patients()
        return [p.to_dict() for p in patients]

    # GET - consultar paciente por id
    @router.get("/patients/{id}")
    def get_patient_by_id(id: int):
        patient = patient_service.get_patient_by_id(id)
        if not patient:
            raise HTTPException(status_code=404, detail="Paciente no encontrado")
        return patient.to_dict()

    # DELETE - para eliminar un paciente
    @router.delete("/patients/{id}")
    def delete_patient(id: int):
        deleted = patient_service.delete_patient(id)
        if not deleted:
            raise HTTPException(status_code=404, detail="Paciente no encontrado")
        return {"message": "Paciente eliminado correctamente"}

    return router
